doc/database_migration.md

```diff
@@ -0,0 +1,206 @@
+init - init migration environment
+revision - create a new migration revision
+commit - merge all the existing revisions and generate a new frozen database version
+upgrade - upgrade the database to the specific revision
+install - generates database set and then installs it in the specified output directory
+revert - execute down.sql script for the newest migration
+redo - execute down.sql and up.sql scripts for the newest migration
+```
+
+All the required info regarding specific subcommands can be printed by passing `--help` as its argument, for instance:
+
+```
+python3 db_migration.py revision --help                            
+usage: db_migration.py revision [-h] [-e ENV] --db DB -m MESSAGE
+
+Creates a new database migration revision
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -e ENV, --env ENV     environment location
+  --db DB               database name
+  -m MESSAGE, --message MESSAGE
+                        revision message
+
+```
+
+## Adding database migration step-by-step
+
+Below, there is a detailed step-by-step instruction on how to add/update database migration:
+
+1. Add build directory to the env variable: `export DB_MIGRATION_ENV=<build_dir>`. This step is optional but highly
+   recommended. Without it, you would need to pass it each time you invoke the migration tool.
+2. Execute `python3 db_migration.py revision --db <database_you_want_to_update> -m "Short description of the revision"`.
+   This will create a `current` folder(if it doesn't exist already) and place a revision entry under it.
+   The revision entry looks like this:
+   ``` 
+   current
+   └── 65a9ee3e_description
+        ├── .meta 
+        ├── devel.sql
+        ├── down.sql
+        └── up.sql
+   ```
+   Its name(`65a9ee3e_description`) is created from the first 8 chars of the assigned unique ID and description string,
+   that you passed to the command. The script also generated a set of empty SQL files: `up.sql`, `down.sql`
+   and `devel.sql`
+   that you will have to fill up. `.meta` file contains various info about revisions like creation date, unique
+   identifier
+   and more. **Do not modify or remove it!**
+3. Add required SQL to `up.sql` and `down.sql`. Remember that `down.sql` should revert all the changes introduced
+   by `up.sql`.
+4. If you do not need development-specific migration, you can remove `devel.sql`.
+5. To apply newly added migration, you can use either:
+    * `python3 db_migration.py upgrade --db <database>`
+    * `product-disk-img` target
+6. You can also upgrade a database to the specific revision by
+   invoking `python3 db_migration.py upgrade --db <database> --rev <revision_id>`. Revision ID can be obtained from
+   the revision's meta file.
+7. Congratulations, you've managed to update the database schema.
+
+## Committing revisions
+
+When the time has come for issuing a new official database release all you have to do is invoke the:
+
+`python3 db_migration.py commit`
```

Comment:
shouldn't it be included in the release process? (e.g. release build on CI?)

Reply:
It would be the best solution. I assume, for now, we would have to do it manually as part of freezing master branch process.